src/actions/matches.js

```diff
@@ -2,48 +2,139 @@ import { database } from './../store';
 import matchModel from './../models/matches';
 
 // set match
+//the match also needs to be added to the 2 users
 export const addMatch = users => {
-    const matchId = database.ref('/matches').push().key;
-    const model = matchModel({
-        competitors: users
-    });
-    return database.ref(`/matches/${matchId}`).set(model);
-}
+  const ref = database.ref('/');
+  const matchId = database.ref('/matches').push().key;
+
+  const model = matchModel({
+    competitors: users
+  });
+  let updateData = {};
+  updateData[`matches/${matchId}`] = model;
+
+  //also update the match data for the competitors
+  users.forEach(userId => {
+    updateData[`users/${userId}/matches/${matchId}`] = true;
+  });
 
+  return ref.update(updateData, function(error) {
+    if (error) {
+      console.log('Error updating data:', error);
+    }
+  });
+};
+
+// set the winner of the match
+// also set the winner/loser with the competitor matches
 export const declareWinner = (matchId, match, winner, declarer) => {
-    const winners = Object.assign({}, match.winners, { [declarer.uid]: winner.uid });
-    match = Object.assign({}, match, { winners });
-    const { competitors, ...update } = setMatchStatus(matchId, match);
-    return database.ref(`/matches/${matchId}`).update(update);
-}
+  const ref = database.ref('/');
+  let updateData = {};
+  const winners = Object.assign({}, match.winners, {
+    [declarer.uid]: winner.uid
+  });
+  match = Object.assign({}, match, { winners });
+  match = setMatchStatus(matchId, match);
 
-export const removeMatch = matchId => {
-    return database.ref(`/matches/${matchId}`).remove();
-}
+  updateData[`matches/${matchId}/winners`] = match.winners;
+  updateData[`matches/${matchId}/isContested`] = match.isContested;
 
-export const setMatchStatus = (matchId, match) => {
-    let winner;
-    let first = true;
-    let isContested = false;
+  if (match.winner) {
+    updateData = Object.assign(
+      {},
+      updateData,
+      updateWinnerData(matchId, match)
+    );
+  }
 
-    // if there are not the same amount of winners as competitors than there is nothing to do
-    if (Object.keys(match.winners).length !== Object.keys(match.competitors).length) {
-        return match;
+  return ref.update(updateData, function(error) {
+    if (error) {
+      console.log('Error updating data:', error);
     }
-    Object.values(match.winners).forEach((value) => {
-        if (first) {
-            winner = value;
-            first = false;
-        }
-        if (winner !== value) {
-            // a contested result
-            isContested = true;
-        }
-    });
+  });
+};
+
+//the match needs to be removed from /matches
+//and the different users
+export const removeMatch = (matchId, match) => {
+  const ref = database.ref('/');
+  let updateData = {};
+  updateData[`matches/${matchId}`] = null;
+  Object.values(match.competitors).map(competitor => {
```

Comment:
reduce

Reply:
you cant reduce an object. updateData is an object.
because firebase does everything with objects

Reply:
Nevermind

Reply:
hehehehe if you can map you can also reduce 🤣